src/workerd/api/streams/standard.c++

```diff
@@ -2252,7 +2252,7 @@ jsg::Promise<void> ReadableStreamJsController::cancel(
 
   const auto doCancel = [&](auto& consumer) {
     auto reason = js.v8Ref(maybeReason.orDefault([&] { return js.v8Undefined(); }));
-    KJ_DEFER(state.init<StreamStates::Closed>());
+    KJ_DEFER(doClose(js));
```

Comment:
Fixing some of the wpt's with the JSG_STRUCT change uncovered a bug here... that fixed other wpt's... yay!